server/command.go

```diff
@@ -264,10 +278,20 @@ func (p *Plugin) executeUnlinkCommand(args *model.CommandArgs) (*model.CommandRe
 		return &model.CommandResponse{}, nil
 	}
 
+	if _, err = p.API.UnshareChannel(link.MattermostChannelID); err != nil {
+		if err2 := p.store.SaveChannelSubscription(*subscription); err2 != nil {
+			p.API.LogDebug("Unable to rollback the subscription deletion", "subscriptionID", subscription.SubscriptionID, "error", err2.Error())
+		}
```

Comment:
See above for probably doing nothing in this case, but even if this was "live", I'm wondering why we would try to rollback the subscription here? Presumably we should still delete the subscription?

Reply:
This is because it is consider that the whole operation has fail, I think this is the correct behavior. You are trying to "unshare" the channel, and because you were unable, you share it again.

Reply:
I'm not sure I'd expect those semantics as an enduser. I agree that in a purely transactional sense, this is ideal -- but the inability to stop sharing a channel in a subsystem internal to us shouldn't preclude disconnecting the MS Teams side. That succeeded and severs the link, even if we have to fix issue internally.

(In other words, I'd rather we harden the code to deal with shared channels that have no matching subscription, than force this error onto the enduser and prevent them from unlinking in the first place.)